node_modules/oae-principals/lib/internal/dao.js

```diff
@@ -32,6 +32,50 @@ var User = require('oae-principals/lib/model').User;
 
 var RESTRICTED_FIELDS = ['acceptedTC', 'admin:tenant', 'admin:global'];
 
+// Users have 24 hours to verify an email address
+var EMAIL_MAPPING_TTL = 24 * 60 * 60;
+
+/**
+ * Create a user record in the database
+ *
+ * @param  {User}       user            The user to persist
+ * @param  {Function}   callback        Standard callback function
+ * @param  {Object}     callback.err    An error object, if any
+ */
+var createUser = module.exports.createUser = function(user, callback) {
+    var queries = [];
+
+    // Persist the user object
+    var values = {
+        'tenantAlias': user.tenant.alias,
+        'displayName': user.displayName,
+        'visibility': user.visibility,
+        'email': user.email,
+        'emailPreference': user.emailPreference,
+        'emailVerified': user.emailVerified,
+        'locale': user.locale,
+        'publicAlias': user.publicAlias,
+        'smallPictureUri': user.picture.smallUri,
+        'mediumPictureUri': user.picture.mediumUri,
+        'largePictureUri': user.picture.largeUri,
+        'acceptedTC': (user.acceptedTC ? user.acceptedTC.toString() : null),
+        'guest': user.guest
+    };
+    queries.push(Cassandra.constructUpsertCQL('Principals', 'principalId', user.id, values));
+
+    // If the user contained an email address, map the email to the user id
+    if (user.email) {
+        var ttl = EMAIL_MAPPING_TTL;
```

Comment:
This makes sense. Might want to add some documentation to explain that the ttl is there to prevent you from "blocking" an email address that isn't used from being used by the rightful owner